mlflow/tracking/fluent.py

```diff
@@ -30,12 +30,13 @@
 from mlflow.utils.validation import _validate_run_id
 from mlflow.utils.annotations import experimental
 
-try:
-    import pandas as pd
+if TYPE_CHECKING:
+    import matplotlib  # pylint: disable=unused-import
+    import plotly  # pylint: disable=unused-import
+    import PIL  # pylint: disable=unused-import
+    import numpy  # pylint: disable=unused-import
+    import pandas  # pylint: disable=unused-import
```

Comment:
This block is not executed at run time: https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING